csm_web/scheduler/tests/models/test_swap.py

```diff
@@ -0,0 +1,233 @@
+import pytest
+import json 
+
+from django.urls import reverse
+from scheduler.models import Swap
+from scheduler.factories import (
+    UserFactory, 
+    CourseFactory, 
+    SectionFactory, 
+    StudentFactory, 
+    MentorFactory
+)
+
+
+@pytest.fixture
+def setup_scheduler(db):
+    """
+    Creates a pair of sections within the same course, each with a mentor and student.
+    """
+    # Setup course
+    course = CourseFactory.create()
+    # Setup sections
+    section_one = create_section(course)
+    section_two = create_section(course)
+    # Setup students
+    section_one_students = create_students(course, section_one, 3)
+    section_two_students = create_students(course, section_two, 3)
+
+    return course, section_one, section_two, section_one_students, section_two_students
+
+
+@pytest.mark.django_db
+def test_basic_swap_request_success(client, setup_scheduler):
+    """
+    Tests that a student can successfully request a swap.
+    """
+    section_one_students, section_two_students = setup_scheduler[3:]
+    sender_student, reciever_student = section_one_students[0], section_two_students[0]
+    # Create a swap request
+    create_swap_request(client, sender_student, reciever_student)
+    # Make sure that the swap request was created
+    assert Swap.objects.filter(receiver=reciever_student).exists()
+    
+
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    ["is_empty"],
+    [
+        # expect empty swap list
+        (True,),
+        # expect non-empty swap list
+        (False,),
+    ],
+    ids=["empty swap list", "non-empty swap list"],
+)
+def test_basic_swap_get(client, setup_scheduler, is_empty):
+    """
+    Tests getting a list of swaps for a student
+    """
+    section_one_students, section_two_students = setup_scheduler[3:]
+    sender_student, reciever_student = section_one_students[0], section_two_students[0]
+    if not is_empty:
+        # Create a swap request
+        create_swap_request(client, sender_student, reciever_student)
+    # Get the list of swaps for the reciever
+    swaps = get_swap_requests(client, sender_student)
+    # Decode get response to UTF-8
+    swaps = json.loads(swaps.content.decode("utf-8"))
+    if not is_empty:
+        # Make sure that the swap request was created
+        assert Swap.objects.filter(receiver=reciever_student).exists()
+        # Make sure that the swap request is in the list of swaps
+        assert len(swaps["sender"]) != 0 and swaps["sender"][0]["receiver"] == reciever_student.id
+    else:
+        assert len(swaps["sender"]) == 0
+
```

Comment:
This should be split up into two tests; the parameterization should be for a single _kind_ of test with various different inputs; here, you're testing two different situations; one where the student has no requests, and one where the student has a swap request.